src/misc.jl

```diff
@@ -30,31 +29,53 @@ function laplacian(con_mat::AbstractMatrix{<:Real})
     return Symmetric(lap, :L)
 end
 
+function laplacian(con_mat::AbstractMatrix{Int})
```

Comment:
Since you're already special casing the `Bool` case:

```julia
function laplacian(con_mat::AbstractMatrix{T}) where T<:Real
    return laplacian(con_mat .> zero(T)) # treating the substation graph as a simple graph.
end
```